File: python/leetcode028.py
class Solution:
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        ls = len(needle)
        l = len(haystack)
        if ls == 0:
            return 0
        for i in range(l-ls+1):
            if haystack[i:i+ls]==needle:
                return i
        return -1

File: leetcodec++/leetcode028.cpp
//
// Created by 黄敏辉 on 2019-05-06.
//

class Solution {
public:
    int strStr(string haystack, string needle) {
        if (needle.size() == 0)
            return 0;
        int res = haystack.find(needle);
        if (res < haystack.size())
            return res;
        else
            return -1;
    }
};